aiokafka/producer/message_accumulator.py

```diff
@@ -236,3 +246,34 @@ def drain_by_nodes(self, ignore_nodes):
         self._wait_data_future = create_future(loop=self._loop)
 
         return nodes, unknown_leaders_exist
+
+    def create_builder(self):
+        magic = 0 if self._api_version < (0, 10) else 1
+        return BatchBuilder(magic, self._batch_size, self._compression_type)
+
+    def add_batch(self, builder, tp):
+        """Add BatchBuilder to queue by topic-partition.
+
+        This method applies no queue limits, so batches may be enqueued up to
```

Comment:
Since the batch interface is intended to grant the user greater control over the data flow and offers an optional flow control mechanism (wait for the batch to be delivered before enqueueing the next or don't), I reasoned that implicit flow control was an unnecessary complication. I'll add it back in.